### ISSUE TITLE
Delimiters and trimming in read_text_dims

### ISSUE THREAD

**Issue by sebpuetz** [finalfusion/finalfusion-rust#38]
I was converting some files from the text(dims) format to our format and ran into trouble with tokens containing whitespaces. There was a number of tokens with non-space whitespaces within tokens which threw our splitting through `.split_whitespace()` off. 

Also, there were a bunch of tokens that were just a whitespace (e.g. NEL `U+0085` and NBSP `U+00A0`), getting reduced to empty strings by the `.trim()` call we do on the words. The second part was particularly frustrating to debug, since `words` ended up with the correct length, the right number of embeddings was created but an incorrect number of vocabulary entries were being created. Which happened because `trim()`ming reduced a couple hundred tokens to `""`.

https://github.com/finalfusion/finalfusion-rust/blob/c9bd67c88c3b4e46bd7a3775271e8d0244e5b643/src/compat/text.rs#L161-L170

This took an awfully long time to debug, mostly due to the stupid format and the unnormalized corpus that the embeddings were trained on, but still, I wonder if we can do better here. E.g.:

```Rust
        let word = parts
            .next()
            .ok_or_else(|| ErrorKind::Format(String::from("Spurious empty line")))?
            .trim_matches(|c| c == ' ' || c == '\t' );
```

What do you think about targeted trimming and splitting on `'\t'` and `'  '` rather than any whitespaces? Might reduce the headaches of potential users.

______

While debugging this, I also found that the following assertion leads to killing the Python process when loading a malformed file through our Python library. https://github.com/finalfusion/finalfusion-rust/blob/c9bd67c88c3b4e46bd7a3775271e8d0244e5b643/src/embeddings.rs#L50-L54

It's kind of ugly to get this kind of output for a malformed file, There's an issue at https://github.com/PyO3/pyo3/issues/492 about stopping panics at the ffi boundary, but for now we would have to deal with that ourselves. Maybe it's possible to return a `Result` here?

```
thread '<unnamed>' panicked at 'assertion failed: `(left == right)`
  left: `199872`,
 right: `200000`: Vocab and norms do not have the same length', /data/rust_projects/finalfusion-rust/src/embeddings.rs:50:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
fatal runtime error: failed to initiate panic, error 5
[1]    8362 abort (core dumped)  ipython
```

**Comment by danieldk** [comment]
Great debugging work!

I am not certain on the last part, in principle `new` is a pure constructor (in the Haskell/FP sense). I think we should keep it that way. If the sizes mismatch, it is a programming error (in this case the reader does not do proper checks).

**Comment by sebpuetz** [comment]
I think this is resolved through #39
